Chauffeur.Tests/Deliverables/DeliveryDeliverableTests.cs

```diff
@@ -290,6 +290,45 @@ public async Task WhenRunWithMultipleParameters_ParametersSubsitutedToDeliverabl
                 .IgnoreAwaitForNSubstituteAssertion();
         }
 
+        [Fact]
+        public async Task WhenRunWithMissingParameters_ErrorsAndOutputsMissingParameters()
+        {
+            var provider = Substitute.For<ISqlSyntaxProvider>();
+            provider.DoesTableExist(Arg.Any<Database>(), Arg.Any<string>()).Returns(true);
+
+            var settings = Substitute.For<IChauffeurSettings>();
+            settings.TryGetChauffeurDirectory(out var s).Returns(x =>
+            {
+                x[0] = @"c:\foo";
+                return true;
+            });
+
+            var cmd = Substitute.For<IDbCommand>();
+            cmd.ExecuteScalar().Returns(1);
+            var conn = Substitute.For<IDbConnection>();
+            conn.CreateCommand().Returns(cmd);
+            var db = new Database(conn);
+
+            var writer = new MockTextWriter();
+
+            var deliverableScript = "foo $test$ $bar$ $baz$";
+            var fs = new MockFileSystem(new Dictionary<string, MockFileData>
+            {
+                {@"c:\foo\bar.delivery", new MockFileData(deliverableScript)}
+            });
+
+            var host = Substitute.For<IChauffeurHost>();
+            host.Run(Arg.Any<string[]>()).Returns(Task.FromResult(DeliverableResponse.Continue));
+
+            var dbSchema = new DatabaseSchemaHelper(db, null, provider);
```

Comment:
Do you actually need to provide the DB into the mock? It shouldn't need it as it shouldn't hit the DB.

Or does it check if the deliverable has been run previously first... I should know the order of execusion...

Reply:
It seems it does hit the database for each delivery to check whether they've been signed for already.